src/Sylius/Component/Channel/Context/ChannelNotFoundException.php

```diff
@@ -15,11 +15,23 @@
 
 class ChannelNotFoundException extends \RuntimeException
 {
+    private static const EXCEPTION_MESSAGE = 'Channel could not be found! Tip: You can use the Web Debug Toolbar to switch between channels in development.';
+
     /**
      * {@inheritdoc}
      */
-    public function __construct(?\Exception $previousException = null)
+    public function __construct($messageOrPreviousException = null, ?\Throwable $previousException = null)
```

Comment:
WDYT about the following implementation:

```php
$message = 'Default message';

if ($messageOrPreviousException instanceof \Throwable) {
  @trigger_error('Passing previous exception as the first argument is deprecated since 1.2 and will be prohibited since 2.0.', E_USER_DEPRECATED);

  $previousException = $messageOrPreviousException;
}

if (is_string($messageOrPreviousException)) {
  $message = $messageOrPreviousException;
}

parent::__construct($message, 0, $previousException);
```

It removes the need for triple parent constructor call and preserves only one execution path (and also trigger the deprecation error).